tests/Unit/Utilities/Test_PrettyType.cpp

```diff
@@ -115,12 +117,319 @@ SPECTRE_TEST_CASE("Unit.Utilities.PrettyType.Stl", "[Utilities][Unit]") {
+  CHECK(pretty_type::short_name<NonTypeType<3, TestType>>() == "NonTypeType");
+  CHECK(pretty_type::short_name<NonTypeType<3, Special>>() == "NonTypeType");
+  CHECK(pretty_type::short_name<NonTypeType<0, int>>() == "NonTypeType");
+  CHECK(pretty_type::short_name<NonTypeType<0, Global>>() == "NonTypeType");
+  CHECK(pretty_type::short_name<NonTypeType<0, TestType>>() == "NonTypeType");
+  CHECK(pretty_type::short_name<NonTypeType<0, Special>>() == "NonTypeType");
+  CHECK(pretty_type::short_name<NonTypeType<-3, int>>() == "NonTypeType");
+  CHECK(pretty_type::short_name<NonTypeType<-3, Global>>() == "NonTypeType");
+  CHECK(pretty_type::short_name<NonTypeType<-3, TestType>>() == "NonTypeType");
+  CHECK(pretty_type::short_name<NonTypeType<-3, Special>>() == "NonTypeType");
+
+  // nullptr-related things
+  CHECK(pretty_type::short_name<Template<std::nullptr_t>>() == "Template");
+  CHECK(pretty_type::short_name<Template2<std::nullptr_t, std::nullptr_t>>() ==
+        "Template2");
+  CHECK(pretty_type::short_name<Template2<int, std::nullptr_t>>() ==
+        "Template2");
+  CHECK(pretty_type::short_name<Template2<Global, std::nullptr_t>>() ==
+        "Template2");
+  CHECK(pretty_type::short_name<Template2<TestType, std::nullptr_t>>() ==
+        "Template2");
+  CHECK(pretty_type::short_name<Template2<Special, std::nullptr_t>>() ==
+        "Template2");
+  CHECK(pretty_type::short_name<Template2<std::nullptr_t, int>>() ==
+        "Template2");
+  CHECK(pretty_type::short_name<Template2<std::nullptr_t, Global>>() ==
+        "Template2");
+  CHECK(pretty_type::short_name<Template2<std::nullptr_t, TestType>>() ==
+        "Template2");
+  CHECK(pretty_type::short_name<Template2<std::nullptr_t, Special>>() ==
+        "Template2");
+  CHECK(pretty_type::short_name<NullptrTemplate<nullptr>>() ==
+        "NullptrTemplate");
+
+  // Enum non-type template parameters
+  CHECK(pretty_type::short_name<
+            GlobalEnumTemplate<Test_PrettyType_Enum::Zero>>() ==
+        "GlobalEnumTemplate");
+  CHECK(pretty_type::short_name<
+            GlobalEnumTemplate<Test_PrettyType_Enum::One>>() ==
+        "GlobalEnumTemplate");
+  CHECK(pretty_type::short_name<
+            Template2<GlobalEnumTemplate<Test_PrettyType_Enum::One>,
+                      GlobalEnumTemplate<Test_PrettyType_Enum::Two>>>() ==
+        "Template2");
+  CHECK(pretty_type::short_name<LocalEnumTemplate<LocalEnum::Zero>>() ==
+        "LocalEnumTemplate");
+  CHECK(pretty_type::short_name<LocalEnumTemplate<LocalEnum::One>>() ==
+        "LocalEnumTemplate");
+  CHECK(
+      pretty_type::short_name<Template2<LocalEnumTemplate<LocalEnum::One>,
+                                        LocalEnumTemplate<LocalEnum::Two>>>() ==
+      "Template2");
+  CHECK(pretty_type::short_name<LocalEnumTemplate<LocalEnum::Negative>>() ==
+        "LocalEnumTemplate");
+
+  // Complicated Test
+  CHECK(pretty_type::short_name<OuterTemplate<
```

Comment:
Technically yes, since `std::string` and `std::nullptr_t` come out as `basic_string` and `decltype(nullptr)`, but I've added a check related to the actual bug with long names.

I also decided to drop the nullptr_t code because I don't think we actually need to get a short name for that.